File: sample-infrastructure-services-domain/dotnet/src/domain-handler.cs
using System;
using System.Threading.Tasks;
using Sample.Services.Abstractions;
using Sample.Domain.Aggregates;

namespace Sample.Infrastructure.Services.Domain
{
    public abstract class DomainHandlerBase<TCommand, TResult> : ICommandHandler<TCommand, TResult> where TCommand : ICommand
    {

        protected readonly ISampleUnitOfWorkFactory _uowFactory;

        public abstract string CommandType { get; }

        public DomainHandlerBase(ISampleUnitOfWorkFactory uowFactory) => _uowFactory = uowFactory;

        public abstract Task<TResult> OnHandleAsync(ISampleUnitOfWork uow, TCommand command);

        public async Task<TResult> HandleAsync(TCommand command)
        {
            TResult result = default(TResult);
            using (var uow = _uowFactory.Create())
            {
                var transactionId = await uow.BeginAsync();
                result = await OnHandleAsync(uow, command);
                await uow.CommitAsync();
            }
            return result;
        }

    }
}

File: sample-infrastructure-services-domain/npm/src/domain-handler.ts
import { injectable } from "inversify";
import { ISampleUnitOfWork, ISampleUnitOfWorkFactory } from "sample-domain";
import { ICommand, ICommandHandler } from "sample-services";

@injectable()
export abstract class DomainHandlerBase<TCommand extends ICommand, TResult=void> implements ICommandHandler<TCommand, TResult> {

    protected readonly _uowFactory: ISampleUnitOfWorkFactory;

    constructor(uowFactory: ISampleUnitOfWorkFactory) {
        this._uowFactory = uowFactory;
    }

    abstract readonly commandType: string;

    protected abstract onHandleAsync(uow: ISampleUnitOfWork, command: TCommand): Promise<TResult>;

    async handleAsync(command: TCommand): Promise<TResult> {
        let uow: ISampleUnitOfWork | null = null;
        let transactionId = "";
        let result: TResult;
        try {
            uow = this._uowFactory.create();
            transactionId = await uow.beginAsync();
            result = await this.onHandleAsync(uow, command);
            await uow.commitAsync();
            await uow.disposeAsync();
            return result;
        }
        catch (e) {
            if (uow) {
                transactionId && await uow.rollbackAsync();
                await uow.disposeAsync();
            }
            throw e;
        }
    }

}